conan/tools/meson/meson.py

```diff
@@ -20,44 +20,46 @@ def configure(self, reconfigure=False):
         has_deps_flags = os.path.exists(deps_flags)
 
         if os.path.exists(cross):
-            cmd += ' --cross-file "{}"'.format(cross)
+            cmd += f' --cross-file "{cross}"'
             if has_deps_flags:
-                cmd += ' --cross-file "{}"'.format(deps_flags)
+                cmd += f' --cross-file "{deps_flags}"'
         else:
-            cmd += ' --native-file "{}"'.format(native)
+            cmd += f' --native-file "{native}"'
             if has_deps_flags:
-                cmd += ' --native-file "{}"'.format(deps_flags)
+                cmd += f' --native-file "{deps_flags}"'
 
-        cmd += ' "{}" "{}"'.format(build_folder, source_folder)
+        cmd += f' "{build_folder}" "{source_folder}"'
         if self._conanfile.package_folder:
-            cmd += ' -Dprefix="{}"'.format(self._conanfile.package_folder)
+            cmd += f' -Dprefix="{self._conanfile.package_folder}"'
         if reconfigure:
-            cmd += ' --reconfigure'
-        self._conanfile.output.info("Meson configure cmd: {}".format(cmd))
+            cmd += f' --reconfigure'
+        self._conanfile.output.info(f"Meson configure cmd: {cmd}")
         self._conanfile.run(cmd)
 
     def build(self, target=None):
         meson_build_folder = self._conanfile.build_folder
-        cmd = 'meson compile -C "{}"'.format(meson_build_folder)
+        cmd = f'meson compile -C "{meson_build_folder}"'
         njobs = build_jobs(self._conanfile)
         if njobs:
-            cmd += " -j{}".format(njobs)
+            cmd += f" -j{njobs}"
         if target:
-            cmd += " {}".format(target)
-        self._conanfile.output.info("Meson build cmd: {}".format(cmd))
+            cmd += f" {target}"
+        self._conanfile.output.info(f"Meson build cmd: {cmd}")
         self._conanfile.run(cmd)
 
     def install(self):
-        self.configure(reconfigure=True)  # To re-do the destination package-folder
+        # To re-do the destination package-folder
+        if self._conanfile.package_folder:
+            self._conanfile.run(f"meson configure --prefix {self._conanfile.package_folder}")
```

Comment:
As far as I can see in the Meson documentation (https://mesonbuild.com/Commands.html#examples), `meson configure` is used to change option values and not to reconfigure everything, isn't it?

Now, I'm running: 

```python
cmd = f'meson configure "{self._conanfile.build_folder}" -Dprefix="{self._conanfile.package_folder}"'
```
So, it should mean that I'm not doing more than changing the `prefix` value. If it does not work like that, is there a way to avoid that full reconfiguration and change only the `prefix` value?

Reply:
Have you tried running it?

...

`meson configure` is used to change option values. It can be run multiple times in a row, does not perform reconfiguration. If the change isn't a no-op, ninja automatically runs `meson setup --reconfigure` to apply the changes.

`meson setup --reconfigure` is used:
- to apply previous updated options
- to force the reconfiguration, even when nothing changed, for example if you've installed new dependencies and suchlike.

> So, it should mean that I'm not doing more than changing the `prefix` value. If it does not work like that, is there a way to avoid that full reconfiguration and change only the `prefix` value?

No.

The full reconfiguration is what processes the changed `prefix` value, optionally enables/disables part of the build or passes `-DPACKAGE_PREFIX=/the/prefix/value` or `configure_file(input: 'constants.py.in', output: 'constants.py', confdata_with_directory_paths)`, or any of various other things that software might do.

The last step of the full reconfiguration is to write out install.dat, which contains the paths that files will be installed to.

For software that isn't relocatable, changing the `prefix` value might force ***all*** source files to be recompiled. Meson cannot know if that is the case until after it does a full reconfiguration and re-creates the ninja file, which may or may not result in changed compilation commands.

Meson does cache what it can (e.g. found dependencies, compiler checks) so the full reconfiguration is often a lot faster than the first time, though it isn't instant.